src/Tools/Source/RunTests/ProcessTestExecutor.cs

```diff
@@ -75,6 +75,8 @@ void MaybeAddSeparator(char separator = '|')
                 builder.AppendFormat($@" --logger {sep}html;LogFileName={GetResultsFilePath(assemblyInfo, "html")}{sep}");
             }
 
+            builder.Append(" --blame-hang-dump-type full --blame-hang-timeout 10minutes");
```

Comment:
Do we also need to pass '--blame-hang'? https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test

```suggestion
            builder.Append(" --blame-hang --blame-hang-dump-type full --blame-hang-timeout 10minutes");
```

Reply:
Nevermind. I see --blame-hang-timeout implies --blame-hang

Reply:
Is the timeout here per-process or per-test? 10 minutes seems pretty low for an entire test partition.

Reply:
Per test.

Reply:
That does get close to a few "stress" tests we've had in the compiler in the past. A 5-7 minute test is not unheard of (don't know if we have any presently though that get that high). A slight machine hiccup could push us to 10 minutes though. 

I'd feel better at 15 but I'm also willing to let 10 ride and see if it has real problems.

Reply:
@jaredpar Sounds good. I picked 10 minutes because I had to pick something, and 10 is a nice number.

Reply:
Switched to 15 minutes. If this causes a problem, it's now @jaredpar's fault.

Reply:
So 15 minutes for a test is not a good setting; why? Because it turns out we also have 15 minutes as our timeout for [the entire Helix job](https://github.com/dotnet/roslyn/blob/33bddf36090b2966f758f27d33b04814d8b429ac/src/Tools/Source/RunTests/TestRunner.cs#L181). @jaredpar @RikkiGibson  given we have that set to 15 that makes me think we don't have any tests getting close to that time; should I bump the Helix job timeout to say 30 minutes or so?

Reply:
(and I've proactively bumped it to 30 minutes to see if something _else_ breaks, but we should continue to discuss)

Reply:
Chatting with @RikkiGibson we're going to go to 10 minute test timeout, 20 for the job. It's not worth us trying to figure out a precise value until we actually have an issue with jobs regularly hitting timeouts and occupying too much time. Our assumption is in practice timeouts are only hit rarely.